### PR TITLE
Feature/google auth

### PR DESCRIPTION
**Pull request by Nikita-Polyakov**
# Task

[PW-XXXX]: Title.

## Changes

1. Description.

[PW-XXXX]: https://soramitsu.atlassian.net/browse/PW-XXXX

### CODE REVIEW

File: src/styles/_mixins.scss [hunk 1]
```diff
@@ -265,3 +265,87 @@
...
+    }
+
+    &__step-count {
+      color: var(--s-color-base-content-secondary);
+      font-size: 14px;
+    }
+  
+    .el-textarea {
+      &__inner {
+        resize: none !important;
```
Comment (stefashkaa) [inline src/styles/_mixins.scss:312]: may we move it somewhere in common styles to avoid `important` here?

File: src/styles/_mixins.scss [hunk 2]
```diff
@@ -265,3 +265,87 @@
...
+
+@mixin drag-drop-content {
+  .placeholder {
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+  }
+
+  &__icon {
+    color: var(--s-color-base-content-tertiary) !important;
```
Comment (stefashkaa) [inline src/styles/_mixins.scss:326]: the same issue - please try to avoid `important` here

File: src/store/transactions/actions.ts [hunk 3]
```diff
@@ -3,15 +3,45 @@ import { Operation } from '@sora-substrate/util';
...
+    );
+  });
+}
+
 const actions = defineActions({
+  async beforeTransactionSign(context): Promise<void> {
+    const { commit, state } = transactionsActionContext(context);
+    const { rootState } = rootActionContext(context);
+
+    if (!rootState.wallet.account.isExternal) {
```
Comment (stefashkaa) [inline src/store/transactions/actions.ts:34]: `if (rootState.wallet.account.isExternal) return;` better than nested conditions

File: src/services/google/wallet/index.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import { addWallet, getWalletBySource } from '@subwallet/wallet-connect/dotsama/wallets';
+
+import { GDriveWalletInfo } from '../../../consts/wallets';
+import { GDriveStorage } from '../index';
+
+import Accounts from './accounts';
+
+class GoogleDriveWallet {
```
Comment (stefashkaa) [inline src/services/google/wallet/index.ts:8]: should it be `@Singletone`?

File: src/services/google/wallet/accounts.ts [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+import { GDriveStorage } from '../index';
+import { api } from '../../../api';
+
+interface IAccountMetadata extends InjectedAccount {
+  id: string;
+}
+
+const ACCOUNTS_UPDATE_INTERVAL = 60_000;
+
+export default class Accounts implements InjectedAccounts {
```
Comment (stefashkaa) [inline src/services/google/wallet/accounts.ts:13]: do we have a reference to this class? DM

File: src/services/google/oauth.ts [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+import { waitForDocumentReady } from '../../util';
+
+type GoogleOauthOptions = {
+  clientId: string;
+  scope: string;
+};
+
+const FIVE_MINUTES = 5 * 60 * 1000;
+
+export class GoogleOauth {
```
Comment (stefashkaa) [inline src/services/google/oauth.ts:11]: should it be `@Singleton`

File: src/services/google/index.ts [hunk 7]
```diff
@@ -0,0 +1,58 @@
+import { GoogleDriveApi } from './api';
+import { GoogleOauth } from './oauth';
+
+const DRIVE_APPDATA_SCOPE = 'https://www.googleapis.com/auth/drive.appdata';
+const DRIVE_DISCOVERY_DOC = 'https://www.googleapis.com/discovery/v1/apis/drive/v3/rest';
+
+class GoogleAccountStorage {
```
Comment (stefashkaa) [inline src/services/google/index.ts:7]: should it be `@Singleton`?

File: src/services/google/api.ts [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+            this._ready = true;
+            resolve();
+          })
+          .catch(reject);
+      });
+    });
+  }
+}
+
+export class GoogleDriveApi extends GoogleApi {
```
Comment (stefashkaa) [inline src/services/google/api.ts:49]: `@Singleton`?

**Review by RustemYuzlibaev — APPROVED** [review]

**Pull request merged**